lib/all_your_base_cases.rb

```diff
@@ -81,28 +81,28 @@ def expected_value
       end
     end
 
-    def invalid_input_digits?
-      row['input_digits'].any? { |x| x < 0 || x >= row['input_base'] }
+    def invalid_digits?
+      row['digits'].any? { |x| x < 0 || x >= row['input_base'] }
     end
 
     def invalid_bases?
       row['input_base'] <= 1 || row['output_base'] <= 1
     end
 
     def input_of_zero?
-      row['input_digits'].all? { |x| x == 0 }
+      row['digits'].all? { |x| x == 0 }
     end
 
     def handle_special_cases
-      [4,2] if row['input_digits'] == [0, 6, 0]
+      [4,2] if row['digits'] == [0, 6, 0]
     end
   end
 end
 
 AllYourBaseCases = proc do |data|
   JSON.parse(data)['cases'].map.with_index do |row, i|
     AllYourBaseCase.new(
-      AllYourBaseCase::PreProcessor.call(row: row).merge(index: i),
+      AllYourBaseCase::PreProcessor[row].merge(index: i),
```

Comment:
Sorry for the confusion... Is this the interface you'd like to see rather than:

```ruby
AllYourBaseCase::PreProcessor.call(row)
```

Thank you so much for the help and feedback!!